exchanges/kraken/kraken_wrapper.go

```diff
@@ -1210,17 +1210,20 @@ func (k *Kraken) GetOrderHistory(ctx context.Context, getOrdersRequest *order.Ge
 			side := order.Side(strings.ToUpper(resp.Closed[i].Description.Type))
 			orderType := order.Type(strings.ToUpper(resp.Closed[i].Description.OrderType))
 			orders = append(orders, order.Detail{
-				ID:              i,
-				Amount:          resp.Closed[i].Volume,
-				RemainingAmount: (resp.Closed[i].Volume - resp.Closed[i].VolumeExecuted),
-				ExecutedAmount:  resp.Closed[i].VolumeExecuted,
-				Exchange:        k.Name,
-				Date:            convert.TimeFromUnixTimestampDecimal(resp.Closed[i].OpenTime),
-				CloseTime:       convert.TimeFromUnixTimestampDecimal(resp.Closed[i].CloseTime),
-				Price:           resp.Closed[i].Description.Price,
-				Side:            side,
-				Type:            orderType,
-				Pair:            p,
+				ID:                   i,
+				Amount:               resp.Closed[i].Volume,
+				ExecutedAmount:       resp.Closed[i].VolumeExecuted,
+				RemainingAmount:      resp.Closed[i].Volume - resp.Closed[i].VolumeExecuted,
+				Cost:                 resp.Closed[i].Cost,
+				CostAsset:            p.Quote,
+				Exchange:             k.Name,
+				Date:                 convert.TimeFromUnixTimestampDecimal(resp.Closed[i].OpenTime),
+				CloseTime:            convert.TimeFromUnixTimestampDecimal(resp.Closed[i].CloseTime),
+				Price:                resp.Closed[i].Description.Price,
+				AverageExecutedPrice: resp.Closed[i].Cost / resp.Closed[i].VolumeExecuted,
```

Comment:
Reference: https://docs.kraken.com/rest/#operation/getOpenOrders

From their example: "OQCLML-BW3P3-BUCMWZ": {
"refid": null,
"userref": 0,
"status": "open",
"opentm": 1616666559.8974,
"starttm": 0,
"expiretm": 0,
"descr": {},
"vol": "1.25000000",
"vol_exec": "0.37500000",
"cost": "11253.7",
"fee": "0.00000",
"price": "30010.0",
"stopprice": "0.00000",
"limitprice": "0.00000",
"misc": "",
"oflags": "fciq",
"trades": []
},

Cost = price * vol_exec, thus can derive cost exec price = cost / vol_exec, and of course also enrich the cost field